<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>733</width>
    <height>560</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Training</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Inputs</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="1">
        <widget class="QLabel" name="lbl_path_target">
         <property name="text">
          <string>Target paths</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QTextEdit" name="te_path_target">
         <property name="toolTip">
          <string>List line per line the paths to the tomograms and their corresponding targets.</string>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOn</enum>
         </property>
         <property name="lineWrapMode">
          <enum>QTextEdit::NoWrap</enum>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.SF NS Text'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;/Users/emoebel/code/python/deep-finder/test/train/in/target0.mrc&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;/Users/emoebel/code/python/deep-finder/test/train/in/target1.mrc&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;/Users/emoebel/code/python/deep-finder/test/train/in/target2.mrc&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="le_path_objl_valid">
         <property name="toolTip">
          <string>The tomogram index in this object list should correspond to the order of above listed tomo/target pairs.</string>
         </property>
         <property name="text">
          <string>/Users/emoebel/code/python/deep-finder/test/train/in/objl_valid.xml</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="lbl_path_objl_train">
         <property name="toolTip">
          <string>The tomogram index in this object list should correspond to the order of above listed tomo/target pairs.</string>
         </property>
         <property name="text">
          <string>Object list (train)</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="lbl_path_objl_valid">
         <property name="toolTip">
          <string>The tomogram index in this object list should correspond to the order of above listed tomo/target pairs.</string>
         </property>
         <property name="text">
          <string>Object list (validation)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QTextEdit" name="te_path_tomo">
         <property name="toolTip">
          <string>List line per line the paths to the tomograms and their corresponding targets.</string>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOn</enum>
         </property>
         <property name="lineWrapMode">
          <enum>QTextEdit::NoWrap</enum>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.SF NS Text'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;/Users/emoebel/code/python/deep-finder/test/train/in/tomo0.mrc&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;/Users/emoebel/code/python/deep-finder/test/train/in/tomo1.mrc&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;/Users/emoebel/code/python/deep-finder/test/train/in/tomo2.mrc&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="le_path_objl_train">
         <property name="toolTip">
          <string>The tomogram index in this object list should correspond to the order of above listed tomo/target pairs.</string>
         </property>
         <property name="text">
          <string>/Users/emoebel/code/python/deep-finder/test/train/in/objl_train.xml</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="lbl_path_tomo">
         <property name="text">
          <string>Tomogram paths</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Training parameters</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="lbl_nclass">
         <property name="toolTip">
          <string>Including background class</string>
         </property>
         <property name="text">
          <string>Number of classes</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="le_nclass">
         <property name="toolTip">
          <string>Including background class</string>
         </property>
         <property name="text">
          <string>3</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="lbl_psize">
         <property name="toolTip">
          <string>Must be a multiple of 4</string>
         </property>
         <property name="text">
          <string>Patch size</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="sb_psize">
         <property name="toolTip">
          <string>Must be a multiple of 4</string>
         </property>
         <property name="minimum">
          <number>40</number>
         </property>
         <property name="maximum">
          <number>80</number>
         </property>
         <property name="singleStep">
          <number>4</number>
         </property>
         <property name="value">
          <number>40</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="lbl_bsize">
         <property name="toolTip">
          <string>Number of patches used to compute average loss</string>
         </property>
         <property name="text">
          <string>Batch size</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="le_bsize">
         <property name="toolTip">
          <string>Number of patches used to compute average loss</string>
         </property>
         <property name="text">
          <string>5</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="lbl_nepochs">
         <property name="toolTip">
          <string>At the end of each epoch, evaluation on validation set is performed (usefull to check if network overfits)</string>
         </property>
         <property name="text">
          <string>Number of epochs</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="le_nepochs">
         <property name="toolTip">
          <string>At the end of each epoch, evaluation on validation set is performed (usefull to check if network overfits)</string>
         </property>
         <property name="text">
          <string>100</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="lbl_steps_per_e">
         <property name="toolTip">
          <string>Number of batches trained on per epoch</string>
         </property>
         <property name="text">
          <string>Steps per epoch</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="le_steps_per_e">
         <property name="toolTip">
          <string>Number of batches trained on per epoch</string>
         </property>
         <property name="text">
          <string>2</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="lbl_steps_per_v">
         <property name="toolTip">
          <string>Number of batches used for validation</string>
         </property>
         <property name="text">
          <string>Steps per validation</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLineEdit" name="le_steps_per_v">
         <property name="toolTip">
          <string>Number of batches used for validation</string>
         </property>
         <property name="text">
          <string>2</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="lbl_direct_read">
         <property name="toolTip">
          <string>Check if you run out of memory when loading dataset. Transmission speed between dataset storage and GPU should be high enough.</string>
         </property>
         <property name="text">
          <string>Direct read</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QCheckBox" name="cb_direct_read">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="toolTip">
          <string>Check if you run out of memory when loading dataset. Transmission speed between dataset storage and GPU should be high enough.</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="lbl_bootstrap">
         <property name="toolTip">
          <string>Can remain checked. Usefull when in presence of unbalanced classes</string>
         </property>
         <property name="text">
          <string>Bootstrap</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QCheckBox" name="cb_bootstrap">
         <property name="toolTip">
          <string>Can remain checked. Usefull when in presence of unbalanced classes</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="lbl_rnd_shift">
         <property name="toolTip">
          <string>(in voxels) Applied to positions in object list when sampling patches. Enhances network robustness. Make sure that objects are still contained in patches when applying shift.</string>
         </property>
         <property name="text">
          <string>Random shift</string>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QLineEdit" name="le_rnd_shift">
         <property name="toolTip">
          <string>(in voxels) Applied to positions in object list when sampling patches. Enhances network robustness. Make sure that objects are still contained in patches when applying shift.</string>
         </property>
         <property name="text">
          <string>13</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QTextEdit" name="te_terminal_out"/>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="gb_out">
      <property name="title">
       <string>Output</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="lbl_path_out">
         <property name="toolTip">
          <string>Path for saving network weights and training metrics</string>
         </property>
         <property name="text">
          <string>Path</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="le_path_out">
         <property name="toolTip">
          <string>Path for saving network weights and training metrics</string>
         </property>
         <property name="text">
          <string>/Users/emoebel/code/python/deep-finder/test/train/out/</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QPushButton" name="button_launch">
      <property name="text">
       <string>Launch</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>733</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
